apps/dc_tools/tests/test_sns_publishing.py

```diff
@@ -0,0 +1,354 @@
+import os
+import json
+import pytest
+import boto3
+from moto import mock_sns, mock_sqs
+from datacube import Datacube
+from click.testing import CliRunner
+from odc.apps.dc_tools.s3_to_dc import cli as s3_cli
+from odc.apps.dc_tools.sqs_to_dc import cli as sqs_cli
+from odc.apps.dc_tools.fs_to_dc import cli as fs_cli
+from datacube import Datacube
+from pathlib import Path
+
+
+@pytest.fixture
+def aws_credentials():
+    """Mocked AWS Credentials for moto."""
+    os.environ["AWS_ACCESS_KEY_ID"] = "testing"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "testing"
+    os.environ["AWS_SECURITY_TOKEN"] = "testing"
+    os.environ["AWS_SESSION_TOKEN"] = "testing"
+
+
+@mock_sns
+@mock_sqs
+def test_s3_publishing_action_from_stac(
+    aws_credentials, aws_env, odc_db_for_sns, s2am_dsid
+):
+    # set up sns topic and the sqs queue that is subscribed to it
+    sns = boto3.client("sns")
+    sqs = boto3.client("sqs")
+    topic_name = "test-topic"
+    queue_name = "test-queue"
+
+    topic = sns.create_topic(Name=topic_name)
+    queue = sqs.create_queue(QueueName=queue_name)
+    attrs = sqs.get_queue_attributes(
+        QueueUrl=queue.get("QueueUrl"), AttributeNames=["All"]
+    )
+    queue_arn = attrs["Attributes"]["QueueArn"]
+    sns_arn = topic.get("TopicArn")
+    sns.subscribe(
+        TopicArn=sns_arn,
+        Protocol="sqs",
+        Endpoint=queue_arn,
+    )
```

Comment:
These lines could go nicely into a fixture, the fixture itself could be annotated with `@mock_sns` and `@mock_sqs`, and could provide the SNS ARN and a [boto3 SQS Queue](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sqs.html#queue) ready to receive messages from.

This would reduce the duplication and make the purpose of the tests easier to follow.

Reply:
Done